protocol/v2/ssv/validator/committee.go

```diff
@@ -98,32 +98,35 @@ func (c *Committee) StartDuty(logger *zap.Logger, duty *spectypes.CommitteeDuty)
 		return errors.New(fmt.Sprintf("CommitteeRunner for slot %d already exists", duty.Slot))
 	}
 
-	validatorShares := make(map[phase0.ValidatorIndex]*spectypes.Share, len(duty.BeaconDuties))
-	toRemove := make([]int, 0)
-	// Remove beacon duties that don't have a share
-	for i, bd := range duty.BeaconDuties {
-		share, ok := c.Shares[bd.ValidatorIndex]
-		if !ok {
-			toRemove = append(toRemove, i)
-			continue
-		}
-		validatorShares[bd.ValidatorIndex] = share
-	}
-	// Remove beacon duties that don't have a share
-	if len(toRemove) > 0 {
-		newDuties, err := removeIndices(duty.BeaconDuties, toRemove)
-		if err != nil {
-			logger.Warn("could not remove beacon duties", zap.Error(err), zap.Ints("indices", toRemove))
-		} else {
-			duty.BeaconDuties = newDuties
-		}
+	//validatorShares := make(map[phase0.ValidatorIndex]*spectypes.Share, len(duty.BeaconDuties))
+	//toRemove := make([]int, 0)
+	//// Remove beacon duties that don't have a share
+	//for i, bd := range duty.BeaconDuties {
+	//	share, ok := c.Shares[bd.ValidatorIndex]
+	//	if !ok {
+	//		toRemove = append(toRemove, i)
+	//		continue
+	//	}
+	//	validatorShares[bd.ValidatorIndex] = share
+	//}
+	//// Remove beacon duties that don't have a share
+	//if len(toRemove) > 0 {
+	//	newDuties, err := removeIndices(duty.BeaconDuties, toRemove)
+	//	if err != nil {
+	//		logger.Warn("could not remove beacon duties", zap.Error(err), zap.Ints("indices", toRemove))
+	//	} else {
+	//		duty.BeaconDuties = newDuties
+	//	}
+	//}
+	//
+	//if len(duty.BeaconDuties) == 0 {
+	//	return errors.New("CommitteeDuty has no valid beacon duties")
+	//}
```

Comment:
we want to filter the shares and execute the committeeRunner only with the relevant shares and duties (these that have a duty in this slot)

Reply:
@GalRogozinski, shouldn't we do changes in spec tests then?

Reply:
@AKorpusenko what test is failing because of that?

Reply:
this will past once https://github.com/ssvlabs/ssv-spec/pull/467 is merged and code is aligned. please add a todo to revert it once its aligned and done.